CriticalMass/MapViewController.swift

```diff
@@ -83,12 +85,43 @@ class MapViewController: UIViewController {
         gpsDisabledOverlayView.set(title: String.mapLayerInfoTitle, message: String.mapLayerInfo)
         gpsDisabledOverlayView.addButtonTarget(self, action: #selector(didTapGPSDisabledOverlayButton))
         view.addSubview(gpsDisabledOverlayView)
+        gpsDisabledOverlayView.addLayoutsSameSizeAndOrigin(in: view)
+        updateGPSDisabledOverlayVisibility()
+    }
+
+    public func presentMapInfo(with configuration: MapInfoView.Configuration) {
+        dismissMapInfo()
+
+        let view = MapInfoView.fromNib()
+        view.configure(with: configuration)
+        self.view.addSubview(view)
+
+        view.translatesAutoresizingMaskIntoConstraints = false
+
+        let topAnchor: NSLayoutYAxisAnchor
+        if #available(iOS 11.0, *) {
+            topAnchor = self.view.safeAreaLayoutGuide.topAnchor
+        } else {
+            topAnchor = self.view.topAnchor
+        }
+
+        let widthLayoutConstraint = view.widthAnchor.constraint(equalTo: self.view.widthAnchor, constant: -32)
+        widthLayoutConstraint.priority = .init(rawValue: 999)
+
         NSLayoutConstraint.activate([
-            gpsDisabledOverlayView.heightAnchor.constraint(equalTo: view.heightAnchor),
-            gpsDisabledOverlayView.widthAnchor.constraint(equalTo: view.widthAnchor),
+            view.topAnchor.constraint(equalTo: topAnchor, constant: 16),
+            widthLayoutConstraint,
+            view.widthAnchor.constraint(lessThanOrEqualToConstant: 400),
+            view.centerXAnchor.constraint(equalTo: self.view.centerXAnchor),
         ])
 
-        updateGPSDisabledOverlayVisibility()
+        UIAccessibility.post(notification: .layoutChanged, argument: view)
+    }
+
+    public func dismissMapInfo() {
+        view.subviews
```

Comment:
Couldn't you use the `gpsDisabledOverlayView ` property directly here without going through the view hierarchy?

Reply:
Yes, you are right, the current solution isn't great . I'm thinking about introducing a MapOverlayViewController in a future PR. to unify the overlay specific code without making the MapViewController more complicated. Especially to make sure that only one overlay is visible at a time etc.